### PR TITLE
Add `MatrixBeta`

### PR DESCRIPTION
**Pull request by johnczito**
Implements the basic features here:

https://en.wikipedia.org/wiki/Matrix_variate_beta_distribution

A couple comments:

- The Wikipedia article draws directly from Chapter 5 of [Gupta/Nagar](https://www.crcpress.com/Matrix-Variate-Distributions/Gupta-Nagar/p/book/9781584880462) (GN). I would normally strive to conform to these sources regarding notation, but instead of saying `U ~ MB(a, b)`, I have opted for `U ~ MB(n₁ / 2, n₂ / 2)`. So the user supplies `n₁` and `n₂`, and the implied `a` and `b` are `a = n₁ / 2` and `b = n₂ / 2`. I do this because this is what practically every source besides GN does: [Mitra](https://www.jstor.org/stable/25049638), [Muirhead](https://www.wiley.com/en-us/Aspects+of+Multivariate+Statistical+Theory-p-9780471769859), etc. Furthermore, I think it is more natural for the user to directly supply the degrees of freedom of the Wishart distributions used for sampling than to supply twice these degrees of freedom.

- I opened JuliaStats/StatsFuns.jl#72 for eventual use in the integrating constant here.

### CODE REVIEW

**Comment by matbesancon** [comment]
@johnczito should we wait for JuliaStats/StatsFuns.jl#72 ?

File: src/matrix/matrixbeta.jl [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+If ``\\mathbf{S}_1\\sim W(n_1,\\mathbf{I}_p)`` and ``\\mathbf{S}_2\\sim W(n_2,\\mathbf{I}_p)``
+are independent, and we use ``\\mathcal{L}(\\cdot)`` to denote the lower Cholesky factor, then
+
+```math
+\\mathbf{U}=\\mathcal{L}(\\mathbf{S}_1+\\mathbf{S}_2)^{-1}\\mathbf{S}_1\\mathcal{L}(\\mathbf{S}_1+\\mathbf{S}_2)^{-\\rm{T}}
+```
+
+has ``\\mathbf{U}\\sim MB(n_1/2, n_2/2)``.
+"""
+struct MatrixBeta{T <: Real, TW <: Wishart} <: ContinuousMatrixDistribution
```
Comment (matbesancon) [inline src/matrix/matrixbeta.jl]: Could we have the field in a more compact form (no blank line in-between)

**Comment by johnczito** [comment]
Right. I wasn't sure what best practices are for that. Even if it gets merged, we still have to wait for a release to use it, right?

File: src/matrix/matrixbeta.jl (hunk 1 above)
Comment (johnczito) [reply src/matrix/matrixbeta.jl]: Done. Thanks!

**Comment by matbesancon** [comment]
yes correct

**Comment by johnczito** [comment]
I don't know how long the wait will be, but it seems unlikely that a badly conflicting commit will force me to reopen this in the meantime, so waiting is fine if you'd prefer.

File: test/matrixbeta.jl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    @test !insupport(C, rand(B))                 #  wrong dims
+    @test !insupport(B, rand(p, p))              #  U isn't posdef
+    @test !insupport(B, rand(B) + I)             #  I - U isn't posdef
+end
+
+@testset "MatrixBeta logpdf" begin
+    @test logpdf(C, Z) ≈ logpdf(c, z)
+end
+
+@testset "MatrixBeta sample moments" begin
```
Comment (matbesancon) [inline test/matrixbeta.jl]: Could we reduce a bit the number of samples?

**Comment by matbesancon** [comment]
@johnczito one minor comment on tests. Afterwards, we can merge this and we'll put a release out, it's been too long

File: test/matrixbeta.jl (hunk 2 above)
Comment (matbesancon) [reply test/matrixbeta.jl]: Think you can still divide by 5

**Comment by johnczito** [comment]
@matbesancon Sounds good. I appreciate your patience with all of these PRs. The feedback has been very helpful.

As for release, the last thing I have for now is a `MatrixFDist` pretty much ready to go, and I've drafted new docstrings for `Wishart` and `InverseWishart` so that they match the standard of these new ones. I can do those very soon or wait until after the release. Whichever you would prefer.

**Comment by matbesancon** [comment]
>  I appreciate your patience with all of these PRs.

You're very welcome, as a non-expert on random matrices, I appreciate the contributions on these areas!

> As for release, the last thing I have for now is a MatrixFDist pretty much ready to go, and I've drafted new docstrings for Wishart and InverseWishart so that they match the standard of these new ones.

Ok sure, we can wait a couple days for these PRs. Especially it would be too bad to release before the doc fix part. For `MatrixFDist` depends how much time you need to polish it

**Pull request merged**